cmd/crictl/main.go

```diff
@@ -55,17 +56,14 @@ func getRuntimeClientConnection(context *cli.Context) (*grpc.ClientConn, error)
 	if RuntimeEndpoint == "" {
 		return nil, fmt.Errorf("--runtime-endpoint is not set")
 	}
-
-	addr, dialer, err := util.GetAddressAndDialer(RuntimeEndpoint)
-	if err != nil {
-		return nil, err
-	}
-
-	conn, err := grpc.Dial(addr, grpc.WithInsecure(), grpc.WithBlock(), grpc.WithTimeout(Timeout), grpc.WithContextDialer(dialer))
-	if err != nil {
-		return nil, errors.Wrap(err, "connect, make sure you are running as root and the runtime has been started")
+	logrus.Debugf("get runtime connection")
+	// As dockershim is deprecated as a default CRI server, it still needs to
+	// be supported. This allows other default endpoint types to be
+	// checked if cannot connect to dockershim
+	if strings.Contains(RuntimeEndpoint, "dockershim") {
```

Comment:
They might not use the exact same path to `[0]` "unix:///var/run/dockershim.sock".. I think you need to try to connect to exactly what they set at the cli or config, if so configured. Suggestion: set a flag to skip trying the defaults if configured..  then here if !configured you can go with the defaults?

Reply:
+1 as well

Reply:
@hickeyma could you address the comments here?

Reply:
@mikebrow @feiskyer I have added a bool to identify if endpoints set by cli flag or config file. I used the original check for "dockershim" as I made the assumption that this would not be set and therefore using the default from https://github.com/kubernetes-sigs/cri-tools/blob/master/cmd/crictl/main.go#L190. This may not be the case as the user may set iits as the endpoint also.

Reply:
Modified.